src/vos/vos_dtx.c

```diff
@@ -4007,27 +4007,44 @@ vos_dtx_local_end(struct dtx_handle *dth, int result)
 	return result;
 }
 
-uint32_t
-vos_dtx_get_cmt_cnt(daos_handle_t coh)
+int
+vos_dtx_get_cmt_cnt(daos_handle_t coh, uint32_t *cnt)
 {
 	struct umem_instance   *umm;
 	struct vos_container   *cont;
 	struct vos_dtx_blob_df *dbd;
-	uint32_t                cnt;
+	uint32_t                tmp;
+	int                     rc;
 
 	cont = vos_hdl2cont(coh);
-	D_ASSERT(cont != NULL);
+	if (cont == NULL) {
+		rc = -DER_INVAL;
+		goto out;
+	}
+	if (cnt == NULL) {
+		rc = -DER_INVAL;
+		goto out;
+	}
 
-	cnt = 0;
+	tmp = 0;
 	umm = vos_cont2umm(cont);
 	dbd = umem_off2ptr(umm, cont->vc_cont_df->cd_dtx_committed_head);
 	while (dbd != NULL) {
-		D_ASSERTF(dbd->dbd_magic == DTX_CMT_BLOB_MAGIC,
-			  "Corrupted committed DTX blob (2) %x\n", dbd->dbd_magic);
-		cnt += dbd->dbd_count;
+		if (dbd->dbd_magic != DTX_CMT_BLOB_MAGIC) {
+			D_ERROR("Committed DTX blob with bad magic: container=" DF_UUID
+				", wait=%#x, get=%#x\n",
+				DP_UUID(cont->vc_id), DTX_CMT_BLOB_MAGIC, dbd->dbd_magic);
+			rc = -DER_INVAL;
+			goto out;
+		}
+		tmp += dbd->dbd_count;
 
 		dbd = umem_off2ptr(umm, dbd->dbd_next);
 	}
 
-	return cnt;
+	*cnt = tmp;
+	rc   = -DER_SUCCESS;
```

Comment:
[*suggestion*] I am wondering why return `-DER_SUCCESS` instead of 0? That seems confused although we usually define `DER_SUCCESS` as 0. You have already described `\return 0 on success, error otherwise` in the header. So please just return 0 and adjust related test cases. Thanks!